storybook/src/data-display/DataTable.stories.tsx

```diff
@@ -796,20 +796,30 @@ export const WithHorizontalScrollAndLastColumnFixed = () => (
   </DataTable>
 );
 
-export const WithDefaultAscendingSortByName = () => (
-  <DataTable
-    data={smallData}
-    defaultSortBy={[
+export const WithDefaultAscendingSortByName = () => {
+  const [dataTableState, dataTableHook] = useDataTable({
+    defaultSortBy: [
       {
         column: "name",
         sortDirection: "ASCENDING",
       },
-    ]}
-  >
-    <DataTable.Column header="ID" accessor="id" canSort={false} />
-    <DataTable.Column header="Name" accessor="name" canSort={true} />
-  </DataTable>
-);
+    ],
+  });
+
+  const sortedData = React.useMemo(() => {
+    return [...smallData].sort((a, b) => {
+      const result = a.name.localeCompare(b.name, undefined, { sensitivity: 'base' });
```

Comment:
Example is too hardcoded and complex. Information we want to send is that there is a sortBy property with a list with column and sortDirection atributes. 

I would rather use something like 
```tsx
const sortInfo = dataTableState.sortBy[0]; // example uses only first sort info
const result = a[sortInfo.column].localeCompare(b[sortInfo.column]);`
return sortInfo?.sortDirection === "DESCENDING" ? -result : result;
```